hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/steps/UserTxn.java

```diff
@@ -263,24 +261,14 @@ public StreamBuilder baseBuilder() {
 
     /**
      * Returns whether the given state indicates this transaction is the first after an upgrade.
+     *
      * @param state the Hedera state
      * @return whether the given state indicates this transaction is the first after an upgrade
      */
     private static boolean isUpgradeBoundary(@NonNull final State state) {
-        final var platformState = state.getReadableStates(PlatformStateService.NAME)
-                .<PlatformState>getSingleton(V0540PlatformStateSchema.PLATFORM_STATE_KEY)
+        final var blockInfo = state.getReadableStates(BlockRecordService.NAME)
+                .<BlockInfo>getSingleton(BLOCK_INFO_STATE_KEY)
                 .get();
-        requireNonNull(platformState);
-        if (platformState.freezeTime() == null
-                || !platformState.freezeTimeOrThrow().equals(platformState.lastFrozenTime())) {
```

Comment:
IIRC, this is being removed so we can depend solely on the `migrationRecordsStreamed` property?

Reply:
True @mhess-swl .